features/device-config/rest/src/main/java/org/opennms/features/deviceconfig/rest/impl/DefaultDeviceConfigRestService.java

```diff
@@ -235,9 +244,38 @@ public Response getDeviceConfigsByInterface(Integer ipInterfaceId, String config
 
     /** {@inheritDoc} */
     @Override
-    public void deleteDeviceConfig(long id) {
-        throw new UnsupportedOperationException("Delete device config not yet supported.");
-        // deviceConfigDao.delete(id);
+    public Response deleteDeviceConfigs(String id) {
+        if (Strings.isNullOrEmpty(id)) {
+            LOG.debug("Bad request : empty or null DeviceConfig Id");
+            return Response.status(Status.BAD_REQUEST).entity("Invalid 'id' parameter").build();
+        }
+
+        var matcher = deletePattern.matcher(id);
+
+        if (!matcher.matches()) {
+            LOG.debug("Bad request : invalid id parameter '{}' ", id);
+            return Response.status(Status.BAD_REQUEST).entity("Invalid 'id' parameter").build();
+        }
+
+        List<Long> ids = Arrays.stream(id.split("\\s*,\\s*"))
+                .filter(s -> !Strings.isNullOrEmpty(s.trim()))
+                .map(Long::parseLong)
+                .collect(Collectors.toList());
+
+        final List<DeviceConfig> deviceConfigList = ids.stream()
+                .map(deviceConfigDao::get)
+                .map(DeviceConfig.class::cast)
+                .collect(Collectors.toList());
+
+        sessionUtils.withTransaction(() -> {
```

Comment:
Fetching an deleting should happen in one transaction.

Reply:
fetching code also moved to into transaction block

Reply:
didn't get ?, currently deleting with one or more id works, its deleting removes record from device-config table in database

Reply:
Currently, you're fetching the full entities form the database just to delete them right afterwards. It may be possible to delete the items in the database just be passing in the IDs without fetching the whole entities first.

Reply:
There is always the way of executing hibernate queries. But this is not performance critical. So this should be fine.